### PR TITLE
Interior via structures

### PR DESCRIPTION
**Pull request by duzos**
This pull request aims to add interiors to the mod via structures

TODO:
- [x] Placing interior 
- [x] Clearing out old interior area
- [x] Bringing in blocks from the outside into the centre of the interior on first generation
- [x] Place blocks from the outside on the first free space above the interior centre
- [x] Fix lag issues with clearing the interior
- [x] Possibly change how interiors are spaced out / generated

### CODE REVIEW

**Comment by duzos** [comment]
I will aim to get this done by the **end of the week** at the latest, but no promises.

**Comment by Plaaasma** [comment]
No rush at all, thank you again for doing this

**Comment by duzos** [comment]
It's just taking a bit because I'm having issues understanding what some of the code does exactly and whats what
I don't mean to be mean but i recommend commenting what certain parts of the code do, and seperating things into different methods to cleanup code

**Comment by duzos** [comment]
I will not be implemeting a way for the player to choose an interior or saving the id of the current interior
I'll leave that to you as its your mod to decide how these features should be implemented and i dont really want to touch the save code lol

**Comment by duzos** [comment]
I believe this all should be fine, let me know if you want any changes, explanations or bug fixes as I dont mind upkeeping this part of the code in the future if need be.

File: src/main/java/net/plaaasma/vortexmod/block/custom/VortexInterfaceBlock.java [hunk 1]
```diff
@@ -201,6 +217,20 @@ public InteractionResult use(BlockState pState, Level pLevel, BlockPos pPos, Pla
                                     newBlockEntity.load(nbtData);
                                     newBlockEntity.invalidateCaps();
                                 }
+
```
Comment (duzos) [inline src/main/java/net/plaaasma/vortexmod/block/custom/VortexInterfaceBlock.java:220]: These comments can be removed

File: src/main/java/net/plaaasma/vortexmod/interior/util/InteriorGenerator.java [hunk 2]
```diff
@@ -0,0 +1,202 @@
...
+        return this.place(
+                server.getLevel(ModDimensions.tardisDIM_LEVEL_KEY),
+                corners[0],
+                corners[1]
+        );
+    }
+
+    /**
+     * Changes the interior
+     * This method
```
Comment (duzos) [inline src/main/java/net/plaaasma/vortexmod/interior/util/InteriorGenerator.java:106]: This javadoc is incorrect

File: src/main/java/net/plaaasma/vortexmod/interior/util/InteriorUtil.java [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+import java.util.Random;
+import java.util.Set;
+
+public class InteriorUtil {
+    public static final Random RANDOM = new Random();
+    public static final int INTERIOR_SIZE = 256;
+
+    /**
+     * Gets the corners of a players interior
+     * TARDIS' will have a size of 10000 in each axis
```
Comment (duzos) [inline src/main/java/net/plaaasma/vortexmod/interior/util/InteriorUtil.java]: incorrect, wont be 10,000 itll be whatever the INTERIOR_SIZE variable is

File: src/main/java/net/plaaasma/vortexmod/interior/util/InteriorUtil.java [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+     */
+    public static boolean isInArea(Entity entity, BlockPos first, BlockPos second) {
+        return isInArea(entity.blockPosition(), first, second);
+    }
+
+    /**
+     * Returns whether a position is within a TARDIS' interior
+     * @param pos the position to look for
+     * @param server server for grabbing location data
+     * @param owner the key in the LocationMapData for a TARDIS ( TODO - I've just realised now that we store interiors using Entity UUIDS, not Owner Hashcodes. Dammit )
```
Comment (duzos) [inline src/main/java/net/plaaasma/vortexmod/interior/util/InteriorUtil.java:94]: Don't worry about the todo though it should be fine and if its not its a quick fix of replacing all the `String owner` with `UUID tardis`

**Pull request merged**